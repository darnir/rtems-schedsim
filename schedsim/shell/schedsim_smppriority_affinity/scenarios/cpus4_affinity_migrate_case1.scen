#
# Create 5 tasks of the same priority.  Then
# change the affinity and the priority of the
# task that is on the ready queue. 
#
rtems_init 4
task_create smp1 4 
task_create smp2 4
task_create smp3 4
task_create smp4 4
task_create smp5 4

# GOAL: want 0x0a010004 - 0x0a010001 as heir and executings
# GOAL:   on CPUs 0 -3 respectively
cpus smp4 smp3 smp2 smp1

task_set_affinity smp5 0x03
task_priority smp5 3
 
# GOAL: want 0x0a010005, 0x0a010003, 0x0a010002,0x0a010001
# GOAL:   as heir and executings on CPUs 0-3 respectively
cpus smp5 smp3 smp2 smp1

task_set_affinity smp4 0x01
# GOAL: want 0x0a010002, 0x0a010005, 0x0a010003,0x0a010001
# GOAL:   as heir and executings on CPUs 0-3 respectively
cpus smp5 smp3 smp2 smp1
