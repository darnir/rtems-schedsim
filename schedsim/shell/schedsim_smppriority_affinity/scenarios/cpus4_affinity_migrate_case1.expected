Script File               : cpus4_affinity_migrate_case1.scen
verbose                   : 1
==> 1: #
==> 2: # Create 5 tasks of the same priority.  Then
==> 3: # change the affinity and the priority of the
==> 4: # task that is on the ready queue. 
==> 5: #
==> 6: rtems_init 4
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
  CPU 1: Thread Heir: 0x09010002 priority=255
  CPU 1: Thread Executing: 0x09010002 priority=255
  CPU 2: Thread Heir: 0x09010003 priority=255
  CPU 2: Thread Executing: 0x09010003 priority=255
  CPU 3: Thread Heir: 0x09010004 priority=255
  CPU 3: Thread Executing: 0x09010004 priority=255
==> 7: task_create smp1 4 
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=4
Task (smp1) starting: id=0x0a010001, priority=4
  CPU 3: Thread Heir: 0x0a010001 priority=4
  CPU 3: Thread Executing: 0x0a010001 priority=4
==> 8: task_create smp2 4
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=4
Task (smp2) starting: id=0x0a010002, priority=4
  CPU 2: Thread Heir: 0x0a010002 priority=4
  CPU 2: Thread Executing: 0x0a010002 priority=4
==> 9: task_create smp3 4
Creating task with:  Preemption: yes Timeslicing: no
Task (smp3) created: id=0x0a010003, priority=4
Task (smp3) starting: id=0x0a010003, priority=4
  CPU 1: Thread Heir: 0x0a010003 priority=4
  CPU 1: Thread Executing: 0x0a010003 priority=4
==> 10: task_create smp4 4
Creating task with:  Preemption: yes Timeslicing: no
Task (smp4) created: id=0x0a010004, priority=4
Task (smp4) starting: id=0x0a010004, priority=4
  CPU 0: Thread Heir: 0x0a010004 priority=4
  CPU 0: Thread Executing: 0x0a010004 priority=4
==> 11: task_create smp5 4
Creating task with:  Preemption: yes Timeslicing: no
Task (smp5) created: id=0x0a010005, priority=4
Task (smp5) starting: id=0x0a010005, priority=4
==> 12: 
==> 13: # GOAL: want 0x0a010004 - 0x0a010001 as heir and executings
==> 14: # GOAL:   on CPUs 0 -3 respectively
==> 15: cpus smp4 smp3 smp2 smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010004 @  4 / 0x0a010004 @  4          false
  CPU 1: 0x0a010003 @  4 / 0x0a010003 @  4          false
  CPU 2: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 3: 0x0a010001 @  4 / 0x0a010001 @  4          false
=== End of Ready Set of Threads
==> 16: 
==> 17: task_set_affinity smp5 0x03
Task (0x0a010005) Set affinity=0x00000003
==> 18: task_priority smp5 3
  CPU 0: Thread Heir: 0x0a010005 priority=3
  CPU 0: Thread Executing: 0x0a010005 priority=3
Task (0x0a010005) Change Priority from 4 to 3
==> 19:  
==> 20: # GOAL: want 0x0a010005, 0x0a010003, 0x0a010002,0x0a010001
==> 21: # GOAL:   as heir and executings on CPUs 0-3 respectively
==> 22: cpus smp5 smp3 smp2 smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010005 @  3 / 0x0a010005 @  3          false
  CPU 1: 0x0a010003 @  4 / 0x0a010003 @  4          false
  CPU 2: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 3: 0x0a010001 @  4 / 0x0a010001 @  4          false
=== End of Ready Set of Threads
==> 23: 
==> 24: task_set_affinity smp4 0x01
Task (0x0a010004) Set affinity=0x00000001
==> 25: # GOAL: want 0x0a010002, 0x0a010005, 0x0a010003,0x0a010001
==> 26: # GOAL:   as heir and executings on CPUs 0-3 respectively
==> 27: cpus smp5 smp3 smp2 smp1
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010005 @  3 / 0x0a010005 @  3          false
  CPU 1: 0x0a010003 @  4 / 0x0a010003 @  4          false
  CPU 2: 0x0a010002 @  4 / 0x0a010002 @  4          false
  CPU 3: 0x0a010001 @  4 / 0x0a010001 @  4          false
=== End of Ready Set of Threads
